redhat-appstudio/remote-secret
# Implement partial update of remote secret data

### What does this PR do?
Implement partial update using get-modify-update approach and configuring the operations by annos on the upload secret.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-561


### How to test this PR?
1. Apply the samples in the following sequence:
   1. `samples/remote-secret.yaml`
   2. `samples/remote-secret-secret.yaml`
   3. `samples/remote-secret-partial-update.yaml`
   4. `samples/remote-secret-partial-delete.yaml`
2. make sure that the secret in the target has the following keys:
   ```
   g: i
   j: k
   ```
